rebar.config

```diff
@@ -10,6 +10,9 @@
     {report_warnings, true},
     {target_erlang_version, 17}
 ]}.
+
+{deps, [{verl, "0.1.0"}]}.
```

Comment:
due to bootstrapping issues in Hex, we cannot use Hex packages in Hex itself and for that reason we can't use them in hex_core too.

Basically, if we want to use verl we'd need to vendor it in `src/*` and update the vendor script to rewrite module names.

We'd definitely like to have semver validation and I'd be happy to use your package (it's great it's Apache 2.0 licensed) but we'd need to vendor it. Could we vendor it as `hex_version`? WDYT @ericmj?

FYI: we vendor hex_core in Hex (as `mix_hex_*`, to avoid conflicting module names, see e.g.: https://github.com/hexpm/hex/blob/master/src/mix_hex_core.hrl) using [`vendor.sh`](https://github.com/hexpm/hex_core/blob/master/vendor.sh)

Reply:
Aye... I just used it to get going... as stated, it's absolutely fine with me if we vendor it in or use something else. It was not clear to me what everyone wanted to do. 

 Also, very much still a WIP : )

Reply:
it's fine to leave this as is until we make a decision which semver parsing solution we'll go with.

Reply:
We could use it into hex_core but we will also need the full feature set from Elixir's `Version` with version comparison and requirement matching, basically it needs to be a 100% reimplementation of `Version`.

Reply:
Yeah, for just validating package version, what verl has is enough. However, eventually we'd also want to validate requirements are in valid format too.

We had conversations about doing registry resolution in hex_core which would indeed need the full feature set of Elixir Version module, but that hasn't been decided yet.

Reply:
Yup, the plan was to carry the rest over. Notice it has the lexer bit, I wanted to do a 0.1.0 with just the parsing then figure out if I needed to get the rest in. It of course is a port of Elixir's Version module, passing with flying colors! But yeah, let me get the rest in first.

Reply:
This is done, 100% (structs and string protocol aside ofc)... Give verl a look over, it should have more coverage than the elixir tests atm, but only slightly, it was well tested in elixir. Let me know how you want to vendor it in.

Reply:
@starbelly verl looks good to me!

Let's vendor it as src/hex_verl*, please include license at the top of the file if it's required (I honestly don't know).

We'd like to have another module that wraps vendored verl. Let's start with minimal module and we'll add more stuff later (e.g. parse_requirement/1 when we actually have such metadata validation)

```erlang
-module(hex_version).
-export([parse/2]).

-type major() :: non_neg_integer().
-type minor() :: non_neg_integer().
-type patch() :: non_neg_integer().
-type pre() :: [binary() | non_neg_integer()].
-type build() :: binary() | undefined.

-type version() :: #{
    major => major(),
    minor => minor(),
    patch => patch(),
    pre   => pre(),
    build => build()
}.

-spec parse(binary()) -> {ok, version()} | {error, invalid_version}.
```

What do you think?

Reply:
I am always a fan of wrapping deps 👍

Reply:
This In the PR now. I did not add the license to the top, they are both apache 2.0, as to whether it _should_ be done anyway, *shrug*